### ISSUE TITLE
[import/no-duplicates] prefer-inline option should work with Flow

### ISSUE THREAD

**Issue by valscion** [import-js/eslint-plugin-import#2686]
Hi!

I noticed that the `prefer-inline` option for `import/no-duplicates` actively forbids using that option with Flow:

https://github.com/import-js/eslint-plugin-import/blob/766af5ff6c2e97cbe2a2a211729f7c129da3005d/src/rules/no-duplicates.js#L119-L123

However, Flow does support inline types [(See this Try flow online REPL)](https://flow.org/try/#0JYWwDg9gTgLgBAJQKYEMDGMA0cDecYCeYScAogDZIhIB28AvnAGZQQhwDkUqGHA3ACgBTAK40MwCDTIAPFOEoAKAJQAuMpWp0APBwAmwAG4cAfLgFw43GCKjTtBwyYASScuQjaA9I5OD6AkA):

```js
import React, { type Element } from 'react';

function Example(): Element<'div'> {
  return <div>Hello</div>;
}
```

So we don't need this:

```js
import React from 'react';
import type { Element } from 'react';

function Example(): Element<'div'> {
  return <div>Hello</div>;
}
```

I spotted this when upgrading from eslint-plugin-import v2.26.0 to v2.27.5 that this old code that works and didn't show lint errors:

```js
import type { Localizable } from 'common/types/locale';
import { localize, localizable } from 'common/types/locale';
import type { FormattedMoney } from 'common/types/money';
import { money, formatMoney, type Money } from 'common/types/money';
```

Now should be written like this instead:

```js
import type { Localizable } from 'common/types/locale';
import { localize, localizable } from 'common/types/locale';
import {
  money,
  formatMoney,
  type Money,
  type FormattedMoney
} from 'common/types/money';
```

Notice how the imports to `common/types/locale`remained the same while the imports to `common/types/money` showed errors.

> ```
>   5:37  warning  '/path/to/common/types/money.js' imported multiple times  import/no-duplicates
>   6:48  warning  '/path/to/common/types/money.js' imported multiple times  import/no-duplicates
> ```

## Expected outcome

I'd expect this to work with Flow:

```js
'import/no-duplicates': ['warn', { 'prefer-inline': true }]
```

## Actual outcome

Error:

> Your version of TypeScript does not support inline type imports. Occurred while linting /path/to/inquiry/utils/priceRangeToText.js:3

**Comment by valscion** [comment]
This requirement to have a specific version of TypeScript installed comes from the original PR by @snewcomer and hasn't been changed since:

* https://github.com/import-js/eslint-plugin-import/pull/2475
